src/cow-react/modules/tokens/hooks/useOnchainBalances.ts

```diff
@@ -0,0 +1,144 @@
+import { Interface } from '@ethersproject/abi'
+import { isAddress } from '@src/utils'
+import { CurrencyAmount, Token } from '@uniswap/sdk-core'
+import JSBI from 'jsbi'
+import ERC20ABI from 'abis/erc20.json'
+import { Erc20Interface } from 'abis/types/Erc20'
+import { useMultipleContractSingleData } from 'lib/hooks/multicall'
+import { ListenerOptionsWithGas } from '@uniswap/redux-multicall'
+import { useMemo } from 'react'
+import { OnchainState } from '../types'
+
+const ERC20Interface = new Interface(ERC20ABI) as Erc20Interface
+const DEFAULT_LISTENER_OPTIONS: ListenerOptionsWithGas = { gasRequired: 185_000, blocksPerFetch: 5 }
+
+export type OnchainTokenAmount = OnchainState<CurrencyAmount<Token> | undefined>
+
+export type OnchainTokenAmounts = {
+  [tokenAddress: string]: OnchainTokenAmount
+}
+
+export type OnchainAmountsResult = {
+  amounts: OnchainTokenAmounts
+  isLoading: boolean
+}
+
+export interface OnchainAmountsParams {
+  account?: string
+  tokens?: (Token | undefined)[]
+  blocksPerFetch?: number
+}
+
+export type OnchainBalancesParams = OnchainAmountsParams
+
+export type OnchainAllowancesParams = OnchainAmountsParams & { spender?: string }
+
+export function useOnchainBalances(params: OnchainBalancesParams): OnchainAmountsResult {
+  const { account } = params
+  const callParams = useMemo(() => [account], [account])
+  return useOnchainErc20Amounts('balanceOf', callParams, params)
+}
+
+export function useOnchainAllowances(params: OnchainAllowancesParams): OnchainAmountsResult {
+  const { account, spender } = params
+  const callParams = useMemo(() => [account, spender], [account, spender])
```

Comment:
Feel free to ignore it

Reply:
I thought so too, but I decided to not change the behavior of this, because I assumed it was an optimization. 
Note that if I don't use the useMemo, it will create a new array instance each time. So I thought that might trigger some additional re-renders. 

I will remove the useMemo just for simplicity, but we might need to revisit the optimizations on this hooks